src/test/java/com/hotels/service/tracing/zipkintohaystack/forwarders/haystack/HaystackDomainConverterTest.java

```diff
@@ -60,21 +81,50 @@ public void shouldConvertZipkinErrorTagIntoSeparateHaystackErrorTags() {
                 .putTag("error", errorMessage)
                 .build();
 
-        Span haystackSpan = HaystackDomainConverter.fromZipkinV2(zipkinSpan);
+        Optional<Span> haystackSpan = victim.fromZipkinV2(zipkinSpan);
 
-        assertTrue(haystackSpan.getTagsList().stream()
+        assertTrue(haystackSpan.isPresent());
+
+        assertTrue(haystackSpan.get().getTagsList().stream()
                 .filter(tag -> "error".equals(tag.getKey()))
                 .map(Tag::getVBool)
                 .findFirst()
                 .orElse(false));
 
-        assertEquals(errorMessage, haystackSpan.getTagsList().stream()
+        assertEquals(errorMessage, haystackSpan.get().getTagsList().stream()
                 .filter(tag -> "error_msg".equals(tag.getKey()))
                 .map(Tag::getVStr)
                 .findFirst()
                 .orElse(null));
     }
 
+    @ParameterizedTest
+    @MethodSource("timestamps")
+    public void shouldDiscardSpanIfTimestampIsInvalid(boolean rejectNullTimestamps, int maxDrift, Long timestamp, boolean spanIsKept) {
```

Comment:
I often try to use real data where possible.. probably something led you to open this issue. If it the data was available it might be telling if it was a bug, or instrumentation that use annotations instead of timestamps, or a race lost (this ran before all the trace ended up here).